DataFormats/QualityControl/src/FlagReasons.cxx

```diff
@@ -10,6 +10,8 @@
 // or submit itself to any jurisdiction.
 
 #include "DataFormatsQualityControl/FlagReasons.h"
+#include "include/DataFormatsQualityControl/FlagReasonFactory.h"
```

Comment:
@aphecetche I have troubles including a header file which is generated by cmake and placed in `"${CMAKE_CURRENT_BINARY_DIR}/include/DataFormatsQualityControl"`. For library source files  the`CMAKE_CURRENT_BINARY_DIR` seems to be added in makefiles, but without the final `include` directory (e.g. `-I/home/pkonopka/alice/O2/cmake-build-debug/DataFormats/QualityControl`), but for test sources it seems it is not added at all.
Could you please give me a hand in fixing this?
@Barthelemy FYI

Reply:
Could you try to see if the version at https://github.com/aphecetche/AliceO2/tree/flag-reasons-csv helps ? (would have done a PR to your branch, but to compile locally I've rebased to the latest dev first ... )
Note that in the `.h` and `.cxx` you _have_ to use `DataFormatsQualityControl/FlagFactory.h` as that's the expected locally when installed (assuming indeed that file is meant for public interface), so the only leeway  is on the `CMakeLists.txt` side.

Reply:
Thank you! It does help for source files, they do not require the hack anymore. However, the tests still do not see the `CMAKE_CURRENT_BINARY_DIR`. For example:
```
cd /home/pkonopka/alice/O2/cmake-build-debug/DataFormats/QualityControl && /home/pkonopka/alice/sw/slc7_x86-64/GCC-Toolchain/v10.2.0-alice2-4/bin/c++ -DBOOST_ALL_NO_LIB -DBOOST_CONTAINER_DYN_LINK -DBOOST_ERROR_CODE_HEADER_ONLY -DBOOST_FILESYSTEM_DYN_LINK -DBOOST_IOSTREAMS_DYN_LINK -DBOOST_PROGRAM_OPTIONS_DYN_LINK -DBOOST_REGEX_DYN_LINK -DBOOST_UNIT_TEST_FRAMEWORK_DYN_LINK -DFAIRMQ_HAS_STD_FILESYSTEM=1 -DFMT_LOCALE -I/home/pkonopka/alice/O2/DataFormats/QualityControl/include -I/home/pkonopka/alice/O2/DataFormats/Headers/include -I/home/pkonopka/alice/O2/DataFormats/MemoryResources/include -I/home/pkonopka/alice/O2/GPU/Common -I/home/pkonopka/alice/O2/Framework/Logger/include -I/home/pkonopka/alice/O2/DataFormats/Detectors/Common/include -I/home/pkonopka/alice/O2/GPU/Utils -I/home/pkonopka/alice/O2/Common/MathUtils/include -I/home/pkonopka/alice/O2/Common/Constants/include -I/home/pkonopka/alice/O2/Utilities/rANS/include -I/home/pkonopka/alice/O2/Common/Utils/include -I/home/pkonopka/alice/O2/DataFormats/common/include -isystem /home/pkonopka/alice/sw/slc7_x86-64/FairMQ/v1.4.43-5/include/fairmq -isystem /home/pkonopka/alice/sw/slc7_x86-64/FairMQ/v1.4.43-5/include -isystem /home/pkonopka/alice/sw/slc7_x86-64/boost/v1.75.0-43/include -isystem /home/pkonopka/alice/sw/slc7_x86-64/FairLogger/v1.10.4-1/include -isystem /home/pkonopka/alice/sw/slc7_x86-64/fmt/8.0.1-1/include -isystem /home/pkonopka/alice/sw/slc7_x86-64/ROOT/v6-24-06-37/include -isystem /home/pkonopka/alice/sw/slc7_x86-64/FairRoot/v18.4.2-47/include -isystem /home/pkonopka/alice/sw/slc7_x86-64/arrow/v5.0.0-alice3-7/include -isystem /home/pkonopka/alice/sw/slc7_x86-64/Vc/1.4.1-74/include -isystem /home/pkonopka/alice/sw/slc7_x86-64/ms_gsl/3.1.0-10/include -fPIC -O2 -std=c++17 -g -O0 -pthread -std=gnu++17 -o CMakeFiles/O2test-dataformatsqualitycontrol-FlagReasons.dir/test/testFlagReasons.cxx.o -c /home/pkonopka/alice/O2/DataFormats/QualityControl/test/testFlagReasons.cxx
/home/pkonopka/alice/O2/DataFormats/QualityControl/test/testFlagReasons.cxx:23:10: fatal error: DataFormatsQualityControl/FlagReasonFactory.h: No such file or directory
   23 | #include "DataFormatsQualityControl/FlagReasonFactory.h"
   ```

Reply:
indeed it does, thank you again.
I will also verify if the header can be accessed from another project (such as QC), but I believe this should be the case.

Reply:
@aphecetche Unfortunately QC cannot find that header even though i put the directory to `PUBLIC_INCLUDE_DIRECTORIES` of that library. Would you happen to know why it could be the case? Could the header be generated after the directory is exported to `PUBLIC_INCLUDE_DIRECTORIES`?

Reply:
@knopers8 I guess the issue is that indeed this file is _not_ installed. Looking at what is installed I see : 

```
📦dev🚀...e/spack/laurent/dev/O2 [flag-reasons-csv] $ tree $(spack location -i o2-aliceo2)/include/DataFormatsQualityControl
/Users/laurent/opt/spack/__spack_path_placeholder__/__spack_path/darwin-monterey-m1/apple-clang-13.0.0/o2-aliceo2-dev-bn2i3zxnnrogyrwk65mylsrzilxhxv2c/include/DataFormatsQualityControl
├── FlagReasonFactory.h.in
├── FlagReasons.h
├── TimeRangeFlag.h
└── TimeRangeFlagCollection.h
```

The fact that the `.in` gets installed is a bit annoying and something that I'll probably have to fix at some point, but the main point is that the `FlagReasonFactory.h` is not installed. Which makes sense as the `o2_add_library` function is only installing what is coming from the source directory. In your case you'll have to explicitly add the relevant `install` command in your `DataFormats/QualityControl/CMakeLists.txt`

Reply:
thank you, I naively though that passing the directories as `PUBLIC_INCLUDE_DIRECTORIES` of the library makes it being done automatically. An `install` command seem to have helped with that.